File: Arrays/Maximum Ascending Subarray Sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int res = 0;
        int cur_sum = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i-1]) {
                cur_sum += nums[i];
            } else {
                res = max(res, cur_sum);
                cur_sum = nums[i];
            }
        }
        return max(res, cur_sum);
    }
};

File: Arrays/Maximum Ascending Subarray Sum.rs
use std::cmp;

impl Solution {
    pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
        let mut res = nums[0];
        let mut cur_sum = nums[0];

        for i in 1..nums.len() {
            if nums[i] > nums[i-1] {
                cur_sum += nums[i];
            } else {
                res = cmp::max(res, cur_sum);
                cur_sum = nums[i];
            }
        }
        cmp::max(res, cur_sum)
    }
}